ddtrace/tracer.py

```diff
@@ -192,6 +193,9 @@ def __init__(
         # globally set tags
         self._tags = config.tags.copy()
 
+        # update git metadata tags
+        self._tags.update(gitmetadata.get_tags())
```

Comment:
We should not be adding this metadata onto every span in the tracer.

Instead we can set it once per-payload and have intake expand this onto every tag.

The reasoning is encoding data is very expensive, adding some additional bytes once per-trace isn't too bad, but once per-span will have a measurable performance overhead.